k8s_state.go

```diff
@@ -0,0 +1,117 @@
+package main
```

Comment:
may be able to use https://pkg.go.dev/tailscale.com/kube:

https://github.com/tailscale/tailscale/blob/main/cmd/containerboot/kube.go#L21-L23

I don't mind client-go tho, it's fine